src/fastoad/module_management/_bundle_loader.py

```diff
@@ -347,39 +372,88 @@ def _get_service_references(
         references = self.framework.find_service_references(service_name, ldap_filter)
         return references
 
-    def _install_python_package(self, package_name: str) -> Tuple[Set[Bundle], Set[str]]:
+    def _install_python_package(self, package_name: str) -> Tuple[Set[Bundle], dict[str, str]]:
         """
-        Recursively loads indicated package.
+        Recursively loads indicated package and its submodules/subpackages.
 
-        :param package_name:
-        :return: A 2-tuple, with the list of installed bundles and the list
-                 of failed modules names
+        :param package_name: Name of the Python package or module to load
+        :return: A 2-tuple:
+            - Set of successfully installed Bundle objects
+            - Dict of failed module names and their full paths
         """
         bundles = set()
-        failed = set()
+        failed = {}  # dict {module_name: full_path}
 
         package = PackageReader(package_name)
+
+        root_package_path = "<unknown>"
+        spec = importlib.util.find_spec(package_name)
+        if spec and spec.origin:
+            root_package_path = str(Path(spec.origin).parent)
```

Comment:
Could you add a small comment about what that does ? And explain it to me at the same time ? 😅